code/API_definitions/session-insights.yaml

```diff
@@ -0,0 +1,792 @@
+          properties:
+            type:
+              enum:
+                - org.camaraproject.session-insights.v0.proposed-resolution
+            data:
+              $ref: "#/components/schemas/ResolutionProposedData"
+
+    SubscriptionEndsNotification:
+      allOf:
+        - $ref: "#/components/schemas/CloudEvent"
+        - type: object
+          properties:
+            type:
+              enum:
+                - org.camaraproject.session-insights.v0.subscription-ends
+            data:
+              $ref: "#/components/schemas/SubscriptionEndsData"
+
+    NotificationEvent:
+      oneOf:
+        - $ref: "#/components/schemas/SessionQualityNotification"
+        - $ref: "#/components/schemas/ScoreAlertNotification"
+        - $ref: "#/components/schemas/ResolutionProposedNotification"
+        - $ref: "#/components/schemas/SubscriptionEndsNotification"
+      discriminator:
+        propertyName: type
+
+    MetricsPayload:
+      type: object
+      description: KPI message published to `metricsTopic`
+      properties:
+        latency:
+          type: number
+          format: float
+        jitter:
+          type: number
+          format: float
+        packetLoss:
+          type: number
+        bitrate:
+          type: number
+        resolution:
+          type: string
+      required:
+        - latency
+        - jitter
+        - packetLoss
+        - bitrate
+        - resolution
+
+    ErrorInfo:
+      type: object
+      properties:
+        status:
+          type: integer
+        code:
+          type: string
+        message:
+          type: string
+      required:
+        - status
+        - code
+        - message
+
+  responses:
+    BadRequest400:
+      description: Bad request
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+
+    Unauthorized401:
+      description: Unauthorized
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+
+    Forbidden403:
+      description: Forbidden
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+
+    NotFound404:
+      description: Not found
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+
+    Conflict409:
+      description: Conflict
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
```

Comment:
I think we also need `410 Gone` (session insights deleted) and the `422` errors for device identifiers.